File: 프로그래머스/lv2/131127. 할인 행사/할인 행사.py
from collections import Counter
def solution(want, number, discount):
    answer = 0
    fruits = {}
    for i in range(len(want)):
        fruits[want[i]] = number[i]
    for i in range(len(discount) - 9):
        check = Counter(discount[i: i+10])
        if fruits == check:
            answer += 1
    return answer

File: 프로그래머스/2/131127. 할인 행사/할인 행사.swift
import Foundation

func solution(_ want:[String], _ number:[Int], _ discount:[String]) -> Int {
    var answer = 0
    var fruits = [String: Int]()
    
    for (index, char) in want.enumerated() {
        fruits[char] = number[index]
    }
    
    for i in 0..<discount.count-9 {
        let subString = discount[i..<i+10]
        
        var check = [String: Int]()
        for char in subString {
            check[char, default: 0] += 1 
        }
        if fruits == check {
            answer += 1
        }
    }
    
    return answer
}